### PR TITLE
TDG post

### PR DESCRIPTION
**Pull request by tomatobrown**
I'm not sure if this is the sort of feel we are going for...can someone take a look at this before the world does

### CODE REVIEW

**Comment by travisemslander** [comment]
@evan-duncan a bit of background here. After visiting Bangladesh I felt a huge thing we could do to help them (and us) recruit good people is getting word about what we're working on out in the public. The team in BD heavily use social networking so I wanted to see if we built up a good cadence of talking about interesting problems and how we're approaching solving them that we can get FN better known in the Dhaka development community.

This eng.fieldnation.com is a good idea Jack had when starting as an engineer at Field Nation, but we forgot about it.

**Comment by tomatobrown** [comment]
also...i'm not totally sure how this Hugo thing works.  Will it just show up if we merge this?  If so, I hope I formatted all of this correctly

File: content/post/test-data-generator.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Our software test should be no different than this.  I’ve seen it time and time again where a test case will run through a handful of tasks, create a user, update that users profile, put the user into some sort of group, and after all that is done then the test will actually start.  This is especially painful with its manual tests doing this because humans have a tendency to be inconsistent with their approach, as well as the fact that its borderline torture to repeat the same tasks like this over and over and over and….you get my point.
```
Comment (travisemslander) [inline content/post/test-data-generator.md]: > especially painful with its manual tests

I think you meant "when it's manual"

File: content/post/test-data-generator.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+We solved this problem at Field Nation using a project that we call the Test Data Generator.  Rather than having a manual test or an automated test click through all the pre-conditions needed to complete the test, we can make a single all to the test-data-generator that will set all of that data to the state we need for the test.  Here is another simple example, let’s say Facebook wants to test that a user can join a group.  If they have the test-data-generator (and I’m sure they have something equivalent) they would have the test began by running something that would generate a brand new user either by directly interacting with the database or via an API, then after the user is created then they could use that user in an integration test and go through the motions of adding the user to the group then checking that it was successfully added to the group.
```
Comment (travisemslander) [inline content/post/test-data-generator.md]: > we can make a single all to the 

single call

Comment (travisemslander) [reply content/post/test-data-generator.md]: > would have the test began by running something

the test begin by

Comment (travisemslander) [reply content/post/test-data-generator.md]: >  an API, then after the user is created then they could 

I'd remove the second then

Comment (tomatobrown) [reply content/post/test-data-generator.md]: fixed I think

**Comment by evan-duncan** [comment]
@tomatobrown I think we might have to do something to publish it with gh-pages but it should be minimal

**Comment by evan-duncan** [comment]
Let’s merge it and see if it auto deploys. If it doesn’t I can deploy it tomorrow.

**Pull request merged**